Tests/ExportersTests/OpenTelemetryProtocol/OtlpHttpMetricsExporterTest.swift

```diff
@@ -0,0 +1,116 @@
+    // The shutdown() function is a no-op, This test is just here to make codecov happy
+    func testShutdown() {
+        let endpoint = URL(string: "http://localhost:\(testServer.serverPort)")!
+        let exporter = OtlpHttpMetricExporter(endpoint: endpoint)
+        XCTAssertNoThrow(exporter.shutdown())
+    }
+    
+    // This test and testGaugeExport() are somewhat hacky solutions to verifying that the metrics got across correctly.  It
+    // simply looks for the metric description strings (which is why I made them unique) in the body returned by
+    // testServer.receiveBodyAndVerify().  It should ideally turn that body into [Metric] using protobuf and then confirm content
+    func testExport() {        
+        let words = ["foo", "bar", "fizz", "buzz"]
+        var metrics: [Metric] = []
+        var metricDescriptions: [String] = []
+        for word in words {
+            let metricDescription = word + String(Int.random(in: 1...100))
+            metricDescriptions.append(metricDescription)
+            metrics.append(generateSumMetric(description: metricDescription))
+        }
+
+        let endpoint = URL(string: "http://localhost:\(testServer.serverPort)")!
+        let exporter = OtlpHttpMetricExporter(endpoint: endpoint)
+        let result = exporter.export(metrics: metrics) { () -> Bool in
+            false
+        }
+        XCTAssertEqual(result, MetricExporterResultCode.success)
+        
+        XCTAssertNoThrow(try testServer.receiveHead())
+        XCTAssertNoThrow(try testServer.receiveBodyAndVerify() { body in
+            var contentsBuffer = ByteBuffer(buffer: body)
+            let contents = contentsBuffer.readString(length: contentsBuffer.readableBytes)!
+            for metricDescription in metricDescriptions {
+                XCTAssertTrue(contents.contains(metricDescription))
+            }
+        })
+        
+        XCTAssertNoThrow(try testServer.receiveEnd())
+    }
+    
+    func testGaugeExport() {
+        let words = ["foo", "bar", "fizz", "buzz"]
+        var metrics: [Metric] = []
+        var metricDescriptions: [String] = []
+        for word in words {
+            let metricDescription = word + String(Int.random(in: 1...100))
+            metricDescriptions.append(metricDescription)
+            metrics.append(generateGaugeMetric(description: metricDescription))
+        }
+        
+        let endpoint = URL(string: "http://localhost:\(testServer.serverPort)")!
+        let exporter = OtlpHttpMetricExporter(endpoint: endpoint)
+
+        let result = exporter.export(metrics: metrics) { () -> Bool in
+            false
+        }
+        XCTAssertEqual(result, MetricExporterResultCode.success)
+        
+        XCTAssertNoThrow(try testServer.receiveHead())
+        XCTAssertNoThrow(try testServer.receiveBodyAndVerify() { body in
+            var contentsBuffer = ByteBuffer(buffer: body)
+            let contents = contentsBuffer.readString(length: contentsBuffer.readableBytes)!
```

Comment:
Hi @vvydier If I remember correctly you know a bit about using protobuf.  Is there any way I can turn the ByteBuffer received here back into [Metric] (or [SpanData] in the OtlpHttpTraceExporterTests)?  Thanks.